CMakeLists.txt

```diff
@@ -23,6 +23,12 @@ endif()
 find_library(LEVELDB_LIBRARY leveldb REQUIRED)
 find_library(NURAFT_LIBRARY nuraft REQUIRED)
 
+message(STATUS "Checking if host system is Apple macOS")
+if (CMAKE_HOST_APPLE)
+ message(STATUS "macOS detected... Disabling Run Time Type Information (RTTI) features.")
+ set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fno-rtti")
```

Comment:
If rtti is no longer needed given the removal of `dynamic_cast` in this PR, I would enable this flag for all platforms rather than just for MacOS.